lib/matplotlib/_constrained_layout.py

```diff
@@ -630,18 +589,16 @@ def layoutcolorbargridspec(parents, cax, shrink, aspect, location, pad=0.05):
         # Horizontal Layout: need to check all the axes in this gridspec
         for ch in gslb.children:
             subspec = ch.artist
-            nrows, ncols, row_start, row_stop, col_start, col_stop = \
-                subspec.get_rows_columns()
             if location == 'right':
-                if col_stop <= maxcol:
```

Comment:
@jklymak btw, are you really sure about whether the various inequalities here must be >/< or >=/<=?  I'm not 100% sure they're all correct, and a comment explaining strict vs non-strict equalities may help...

Reply:
Well, you are the one who decided it should all be off by one, so I assumed you carefully checked ;-) It passes the tests so that is encouraging.  There are lots of quite complex tests, so its hard to see how you broke something if it passes.

Reply:
I actually was referring to the original version.

Reply:
ha ha, OK.  But again, the tests are pretty complex and don't fail in any weird ways.  If these inequalities were not correct, its a little hard to see how the whole thing would work.  But I can go through and try and remember why each is the way it is.

Reply:
I was slightly puzzled by them, so if you can add a comment that would be nice, but there's no urgency whatsoever.

Reply:
These are all correct.  Lets say the parents of the colorbar span columns 1-3 of a six column layout (indexed 0, 1, 2, 3, 4, 5).  `maxrow` is then 3.  If a subplot span stops at 3 or less then the colorbar is to the right (I assume in this case that `subspec.colspan.stop == 4`).  If the subplot *starts* 4 or higher the colorbar is to its left.  

Note its entirely possible for a subplot to not be positioned relative to the colorbar.  i.e. if in another row it spans columns 0-5 that is perfectly allowed and neither stacking would happen.  Saying that I wonder if I *do* have a bug in here where I should only call `hstack` if order is not empty.    But maybe this should go in first and I fix things like that.

Reply:
Hmm, I guess I'll just trust you on this for now and leave things as they are (with the mechanical translation) ;)

Reply:
I think thats fine, but I will go through and fix this after your translation is in.